packages/block-renderer/src/components/pattern-renderer.tsx

```diff
@@ -33,17 +37,19 @@ const PatternRenderer = ( {
 		patternHtml = transformHtml( patternHtml );
 	}
 
-	let patternStyles = pattern?.styles ?? [];
+	let patternStyles = pattern?.styles ? [ ...styles, ...pattern.styles ] : [ ...styles ];
 	if ( shouldShufflePosts ) {
 		const css = shufflePosts( patternId, patternHtml );
 		patternStyles = [ ...patternStyles, { css } as RenderedStyle ];
 	}
 
+	const patternScripts = [ pattern?.scripts ?? '', scripts ];
```

Comment:
`pattern?.scripts` looks like also always exists as `string`.

If you haven't faced the case when `script` or `styles` are `undefined` - I would avoid adding fallback just in case.
IMO we should either avoid fallbacks or fix types (at least add comment if fixing types makes extra effort).

Reply:
It's not the `styles` and `scripts` properties that are potentially undefined, it's `pattern`.